tools/generators/pbxproject_prefix/src/Generator/Generator.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import PBXProj
+
+/// A type that generates and writes to disk a `PBXProject` prefix `PBXProj`
+/// partial.
+///
+/// The `Generator` type is stateless. It can be used to generate multiple
+/// partials. The `generate()` method is passed all the inputs needed to
+/// generate a partial.
+struct Generator {
```

Comment:
>It can be used to generate multiple partials.

Multiple partials of the same `pbxproject_prefix` "type" I assume correct? Given the `generate()` impl it looks like this is specific to `pbxproject_prefix`. 

Are we planning to implement a `Generator` class like this for all partial generators?

Reply:
In practice you wouldn't use the same instance of `PBXProjectPrefix.Generator` to generate multiple partials, but it's stateless, so you could.

> Are we planning to implement a `Generator` class like this for all partial generators?

Yes. Each generator tool will have its own `Generator` struct.

Reply:
Any value in making these `Generator` structs adopt a protocol so we know "a partial Generator" is expected to be able to "init with env, generate, <maybe more expected actions in the future?>" 

Probably not something for this PR, though, just wondering if it could end up evolving this way and requiring a common interface

Reply:
I personally don't see a value in that. There is only a separation between `PBXProjectPrefix` and `Generator`, because we can't set `environment` in `PBXProjectPrefix.run()`. The slip makes it so `PBXProjectPrefix` parses the arguments, then those arguments get passed to `Generator`. The environment and arguments will be tied to each generator. Each command (`PBXProjectPrefix.run` in this case) will work with exactly one generator, so no need for an interface. Also, the command and generator will be the only ones in their respective module/tool.

So yes, there will be multiple commands and generators, but that's just because there will be multiple tools. Each tool could technically do whatever it wants though, it's just that we will be following the pattern set here as a common architecture.